apps/aecore/src/txs/aec_tx.erl

```diff
@@ -40,29 +40,57 @@ fee(#spend_tx{fee = F}) ->
     F.
 
 -spec apply_signed(list(signed_tx()), trees(), non_neg_integer()) ->
-                          {ok, trees()} | {error, term()}.
```

Comment:
This function `aec_tx:apply_signed` changed its API to always succeed by silently discarding invalid txs. I see this same function is used by the module [building the state trees in the chain](https://github.com/aeternity/epoch/blob/b0950fb9b0ddef9653a720486627c09ffff17c8c/apps/aecore/src/aec_chain_state.erl#L530) that shall see application of all txs succeed.

Please separate function in two:
* Once expecting all txs to be valid, and returning error otherwise (I envisage this to be used by (1) module processing block received from network and (2) module building state trees in chain);
* One silently discarding transactions (for mining - as proposed in this PR).

Reply:
I believe that the fact that txs got applied on state tree without failures should be verified by checking hash of state tree (`#block.root_hash` field) when absorbing a block. If a tx cannot be applied, hash simply will not match. And checking `root_hash` is the last thing to be implemented for `PT-#152481000 Validate received block`. I'm not sure if separating this function brings much value then, let me think about it.